Warm-up/DataProc.py

```diff
@@ -0,0 +1,198 @@
+
+    x = len(entersfib)
+    y = len(leavesfib)
+
+    for k in range(0, max(x,y)):
+        try:
+            is_fibrillating.extend([1 for i in range(entersfib[k],leavesfib[k])])
+            is_fibrillating.extend([0 for j in range(leavesfib[k],entersfib[k])])
+        except IndexError:
+            if is_fibrillating[-1] == 1:
+                is_fibrillating.extend([0 for j in range(leavesfib[k],int(params['tmax']))])
+            else:
+                is_fibrillating.extend([1 for i in range(entersfib[k],int(params['tmax']))])
+        	break
+
+    return np.mean(is_fibrillating)
+	
+def risk_curve_plot():
+
+    dirname = 'Data-{0}-{1}-Risk_Curve'.format(params['d'], params['e'])
+    nus = pickle.load(open(dirname + '/Nus-{0}'.format(params['tmax']), "r"))
+
+    mean_time_in_AF = []
+    std_devs = []
+
+    for nu in nus:
+    	time_in_AF = []
+
+    	for i in range(0, params['realisations']):
+    		try:
+    			activity = pickle.load(open(dirname + '/Run-{0}-{1}-{2}'.format(params['tmax'], nu, i), "r"))
+    			is_fibrillating = np.array(activity[0]) >= (params['tissue_shape'][0]+0.05*params['tissue_shape'][0])
+    			time_in_AF.append(float(np.sum(is_fibrillating))/ params['tmax'])
+    		except:
+    			Exception("Data for realisation %i does not exist"%(i))
+    	mean_time_in_AF.append(np.mean(time_in_AF))
+    	std_devs.append(np.std(time_in_AF))
+
+    kishanNus = [0.02, 0.04, 0.06, 0.08, 0.11, 
+                 0.13, 0.15, 0.17, 0.19, 0.21, 
+                 0.23, 0.25, 0.27, 0.29, 0.12, 
+                 0.14, 0.16, 0.18, 0.2, 0.22, 
+                 0.24, 0.26, 0.28, 0.3, 0.1]
+    
+    kishanMeanAF = [0.99981, 0.99983,0.9998,0.99968, 0.99772, 0.96099, 
+                    0.60984, 0.16381, 0.017807, 0.020737, 4.922e-05, 0.0001084,
+                    0,0, 0.99152, 0.86184, 0.29714, 0.039206, 0.0056277,
+                    4.834e-05, 0.00082172, 0,0,9.406e-05, 0.99919]
+    
+    kishanStdDevs = [4.3015e-06, 3.8088e-06, 1.0454e-05, 3.0663e-05, 0.00044859, 
+                     0.018246, 0.054379, 0.041092, 0.0080603, 0.016513, 4.8685e-05, 
+                     8.4968e-05, 0, 0, 0.0027053, 0.028043, 0.055185, 0.013863,
+                     0.0028284, 3.6005e-05, 0.00081342, 0, 0, 9.3115e-05, 0.00010423]
+
+    plt.errorbar(kishanNus,kishanMeanAF, yerr = kishanStdDevs, xerr=None, fmt='x', label = 'Data from Kishan')
+
+    analytic = [(1 - (1 - (1 - nu)**params['refractory_period'])**(params['d'] * params['tissue_shape'][0]**2)) for nu in nus]
+    plt.plot(nus, analytic, '--', label = 'Analytic')
+
+    plt.errorbar(nus, mean_time_in_AF, yerr=[i/(params['realisations'])**0.5 for i in std_devs], xerr=None, fmt='x', label = 'Data from us')
+
+    plt.xlabel('Fraction of vertical connections, '+r'$\nu$')
+    plt.ylabel('Mean time in AF/ Risk of AF')
+    plt.title('Realisations: {0}'.format(params['realisations']))
+    plt.legend()
+    plt.show()
+	
+def plot_activity(nu, i):
```

Comment:
Same again here - we don't really need to keep track of/share scripts for loading and plotting data